darshan-runtime/lib/darshan-stdio.c

```diff
@@ -228,11 +228,11 @@ extern int __real_fileno(FILE *stream);
     __rec_ref = darshan_lookup_record_ref(stdio_runtime->rec_id_hash, &__rec_id, sizeof(darshan_record_id)); \
     if(!__rec_ref) __rec_ref = stdio_track_new_file_record(__rec_id, __newpath); \
     if(!__rec_ref) { \
-        if(__newpath != (char*)__path) free(__newpath); \
```

Comment:
Oh, now that Wei-keng pointed out, I agree that the original logic was intentional (__newpath may point to __path or be a freshly allocated buffer depending on preceding logic).  There is probably another fix for the clang warning?